FeedStoreChallenge/CoreDataFeedStore.swift

```diff
@@ -29,14 +29,61 @@ public final class CoreDataFeedStore: FeedStore {
 	}
 	
 	public func retrieve(completion: @escaping RetrievalCompletion) {
-		fatalError("Must be implemented")
+		let context = self.context
+		context.perform {
+			do {
+				if let cache = try CoreDataFeed.find(in: context) {
+					completion(.found(
+						feed: cache.items
+							.compactMap { ($0 as? CoreDataFeedImage)?.toLocal() },
+						timestamp: cache.timestamp))
```

Comment:
You can move the `compactMap` into a helper in the `cache` class.

Something like `cache.localItems`:

```suggestion
					completion(.found(feed: cache.localItems, timestamp: cache.timestamp))
```